resources/mediaplayer/src/main/java/org/mobicents/media/server/impl/resource/mediaplayer/audio/CachedRemoteStreamProvider.java

```diff
@@ -27,30 +27,58 @@
     public CachedRemoteStreamProvider(int size) {
         cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
                 .withCache("preConfigured",
-                        CacheConfigurationBuilder.newCacheConfigurationBuilder(URL.class, byte[].class,
+                        CacheConfigurationBuilder.newCacheConfigurationBuilder(URL.class, AudioStreamCache.class,
                                 ResourcePoolsBuilder.newResourcePoolsBuilder().heap(size, MemoryUnit.MB))
                                 .build())
                 .build(true);
     }
 
-    public Cache getCache() {
-        return cacheManager.getCache("preConfigured", URL.class, byte[].class);
+    private Cache<URL, AudioStreamCache> getCache() {
+        return cacheManager.getCache("preConfigured", URL.class, AudioStreamCache.class);
     }
 
     public InputStream getStream(URL uri) throws IOException {
-        Cache<URL, byte[]> cache = getCache();
+        Cache<URL, AudioStreamCache> cache = getCache();
 
         if (!cache.containsKey(uri)) {
```

Comment:
doesn't cache support putIfAbsent? That would be preferred to locking. It should also be OK now since stream is now lazily loaded.